content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,170 @@
+---
+title: 'Prisma Client extensions'
+metaTitle: 'Prisma Client extensions (Preview)'
+metaDescription: 'Extend the functionality of Prisma Client'
+preview: true
+tocDepth: 3
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+This page is in draft.
+
+</Admonition>
+
+Use Prisma Client extensions to add functionality to the models in your schema, results objects, queries, or Prisma Client itself.
+
+There are four types of extension. You can use each type to customize a different aspect of Prisma Client.
+
+- `model` extensions: [add custom methods or fields to your models](/concepts/components/prisma-client/client-extensions/client-extensions-model)
+- `client` extensions: [add client-level methods to Prisma Client](/concepts/components/prisma-client/client-extensions/client-extensions-client)
+- `result` extensions (for future release): [add custom results objects to Prisma Client](/concepts/components/prisma-client/client-extensions/client-extensions-result)
+- `query` extensions (for future release): [create custom Prisma Client queries](/concepts/components/prisma-client/client-extensions/client-extensions-query)
+
+</TopBlock>
+
+## About Prisma Client extensions
+
+To create Prisma Client extensions, use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods), and the appropriate extension type (`model`, `client`, `query` or `result`).
+
+You can use [multiple independent extensions at the same time](#multiple-extensions), each in is own isolated instance of Prisma Client.
+
+You can also [share your Prisma Client extensions](#share-extensions) with other Prisma users, and [import Prisma Client extensions developed by other users](#install-a-packaged-extension) into your Prisma project.
+
+## Prerequisites
+
+To use Prisma Client extensions, you must enable the `clientExtensions` feature flag in the generator block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Multiple extensions
+
+Each time that you call `$extends`, you create an _extended client_. This extended client is a forked state of the standard Prisma Client. The original client does not change.
+
+Every extended client operates independently, in an isolated instance. It cannot conflict with other extended clients or the original client.
+
+You can also combine two or more extensions into one extension, if you want the functionality from both extensions to apply in the same client instance.
```

Comment:
You can also "stack" extensions I think? So `prisma` becomes `prismaWithA` and then `prismaWithAandB`.

Reply:
I've asked in the channel about this

Reply:
OK, I have info about this now, so I'll write this up.